src/coprocessor/select/select.rs

```diff
@@ -129,109 +122,30 @@ impl<'a> SelectContext<'a> {
     }
 
     fn get_rows_from_range(&mut self, range: KeyRange) -> Result<usize> {
-        let mut row_count = 0;
-        if is_point(&range) {
+        if self.req_ctx.table_scan && is_point(&range) {
```

Comment:
Why check `self.req_ctx.table_scan`?

Reply:
Because if the range is on a secondary index, there will be several records with same index prefix, in which case we can't deal with it as `point query`. For example:
```sql
create table x (
    a int primary key,
    b int,
    index (b)
);

select * from x where b = 3;
```

Reply:
This should be push down as `IndexScan` rather than `TableScan`. And `get_rows_from_range` will only be called when it's `TableScan`. Otherwise `get_idx_row_from_range` will be called.